chain/jsonrpc/openapi/src/main.rs

```diff
@@ -85,6 +80,362 @@ impl schemars::transform::Transform for ReplaceNullType {
     }
 }
 
+/// This struct is used to interchange `oneOf` and `allOf` in the OpenAPI schema. And also adds titles to the `allOf` schemas.
+/// If you have allOf [oneOf[A, B], oneOf[C, D]] it will transform it into: oneOf [allOf[A, C], allOf[A, D], allOf[B, C], allOf[B, D]].
+///
+/// For example, it will transform the following schema:
+/// ```json
```

Comment:
Can this example be minimized? Reduce the number of properties, maybe make the json formatting more compact?

Reply:
While at it, might as well use the doctest infrastructure to not only give an example but also test its validity.

Reply:
I tried to minimize. All the other fields are needed to show what's going on. 
Doctest is only for libraries, not binaries. I would convert it to 2 packages as soon as it's only for testing if it's a correct json

Reply:
Huh, you're right.